app/src/main/java/org/example/app/DirectoryInfo.java

```diff
@@ -27,6 +27,18 @@ public static Optional<DirectoryInfo> of(Path path) {
         return of(path.toString());
     }
 
+    /**
+     * Returns information about the subdirectories of the current directory.
+     * @return A Set containing information about the subdirectories.
+     */

```

Comment:
Good, this is enough. No need to generate docs, only comments on top of the method.